File: backend/server/src/services/icon_registry.rs
pub const APPROVED_ICON_KEYS: &[&str] = &[
    "layout-dashboard",
    "folder",
    "file-text",
    "sticky-note",
    "calendar-days",
    "clipboard-list",
    "columns",
    "git-branch",
    "share-2",
    "lock",
    "globe",
    "settings",
];

pub fn is_approved_icon_key(icon: &str) -> bool {
    APPROVED_ICON_KEYS.contains(&icon)
}

File: frontend/src/lib/applications/iconRegistry.ts
export const APPROVED_MODULE_ICONS = [
	'layout-dashboard',
	'folder',
	'file-text',
	'sticky-note',
	'calendar-days',
	'clipboard-list',
	'columns',
	'git-branch',
	'path-separation',
	'share-2',
	'lock',
	'globe',
	'settings',
	'lightbulb',
	'activity',
	'mail',
	'message-circle'
] as const;

export const DEFAULT_MODULE_ICON = 'folder';

export type ApprovedApplicationIcon = (typeof APPROVED_MODULE_ICONS)[number];

export function isApprovedApplicationIcon(icon: string): icon is ApprovedApplicationIcon {
	return (APPROVED_MODULE_ICONS as readonly string[]).includes(icon);
}
